### ISSUE TITLE
Make blocklist optional for the user

### ISSUE THREAD

**Issue by alxlg** [tootsuite/mastodon#13483]
I think the way Mastodon network started (as a set of instances with hundreds to thousands of users) doesn't fit with the way blocklists were implemented later.

You may found at a certain point that the admins of the instance you use added a certain domain to their blocklist and you realized you can't interact with regions of the Fediverse. You have to change instance as many points out, but the problem remains on the other direction: the instance you are in may be blocked by other instances where there are users that would be interested in interacting with you but they may not even know you exist or that a blocklist exists.

The problem here is that Mastodon instances are way too large in terms of users to enforce such blocklist system without fragmenting the Fediverse, something that it's happening  because many major instances already have blocked others because of ideological reasons.

I think it's very important for the health of the Mastodon network (and the Fediverse as a consequence) to let the users disable the instance block list for themselves with an option in the settings.

In addition, in my opinion, due to the current abuse of the blocklists, users (existing and new ones) should be prompted with a dialog to let them accept or decline the blocklist by their instance and be informed that there is an option in the settings to change the behavior later.

**Comment by trwnh** [comment]
Duplicate of #11218
